Course work1 hard competed

iShantaram opened the pull request.

iShantaram committed:
Course work 1 on easy mode completed.

iShantaram committed:
Course work 1 on hard mode completed.

iShantaram committed:
Добавлены проверки на 0, при рассчетах avgSalary.
Переименованы методы, возвращающие сотрудников с крайними зарплатами.

iShantaram committed:
Переименованы методы в соответствии с концепцией "начало названия метода - глагол".

DmitryN10 commented on src/pro/sky/java/course1/courseWork1Easy/CourseWork1Easy.java:
```diff
@@ -0,0 +1,88 @@
+
+    public static void printData(Employee[] employees) {
+        for (Employee employee : employees) {
+            if (employee != null) {
```
молодец

DmitryN10 commented on src/pro/sky/java/course1/courseWork1Easy/CourseWork1Easy.java:
```diff
@@ -0,0 +1,88 @@
+                if (employees[j] != null) break;
+                j++;
+            }
+            for (int i = j+1; i < employees.length; i++) {
```
j =количеству не null ячеек в массиве

iShantaram replied:
нет, j - первая не null ячейка.

DmitryN10 commented on src/pro/sky/java/course1/courseWork1Easy/CourseWork1Easy.java:
```diff
@@ -0,0 +1,88 @@
+                if (employees[j] != null) break;
+                j++;
+            }
+            for (int i = j+1; i < employees.length; i++) {
```
почему начинаешь не с начала массива итерироваться. Можешь обьяснить

iShantaram replied:
поэтому начинаю сравнивать зарплату j-й ячейки со следующими.

DmitryN10 commented on src/pro/sky/java/course1/courseWork1Hard/CourseWork1Hard.java:
```diff
@@ -0,0 +1,249 @@
+    }
+
+    private static Employee findMaxSalaryEmployee(int department) {
+        if (Employee.counter > 0) {
```
Я бы так переиспользовал код и в других методах аналогично

Employee[] departEmployees = findEmployeesInDepart(department);
return findMaxSalaryEmployee(departEmployees);

iShantaram replied:
вроде переделал, но немного сомневаюсь, не накосячил ли я с логикой)

DmitryN10 commented on src/pro/sky/java/course1/courseWork1Easy/CourseWork1Easy.java:
```diff
@@ -0,0 +1,88 @@
+        }
+    }
+
+    public static Employee maxSalary(Employee[] employees) {
```
так можно


```
    public static Employee calculationMaxSalary(Employee[] arrayEmployees) {
        Employee result = null;
        double maxSalary = Integer.MIN_VALUE;
        for (Employee employee : arrayEmployees) {
            if (employee != null && employee.getSalary() > maxSalary) {
                maxSalary = employee.getSalary();
                result = employee;
            }
        }
        return result;
    }
```

iShantaram replied:
да уж, так гораздо изящнее!  Переделал.

iShantaram committed:
Вовсе методы добавлена передача массива сотрудников.
Добавлен метод поиска сотрудников в конкретном отделе.
Поэтому теперь не нужны отдельные методы для "сотрудников отдела", можно вызвать любой метод, указав сотрудников требуемого департамента.
Переделаны методы поиска сотрудника с минимальной и максимальной зарплатой.

DmitryN10 commented on src/pro/sky/java/course1/courseWork1Hard/CourseWork1Hard.java:
```diff
@@ -0,0 +1,186 @@
+                countEmployees++;
+            }
+        }
+        Employee[] result = new Employee[countEmployees];
```
все верно делаешь
